source/MissionAction.cpp

```diff
@@ -419,6 +431,25 @@ bool MissionAction::CanBeDone(const PlayerInfo &player, const shared_ptr<Ship> &
 	
 	for(const auto &it : requiredOutfits)
 	{
+		// Special case: map
+		int mapSize = it.first->Get("map");
+		if(mapSize > 0)
+		{
+			if(it.second == 0)
+			// Condition is to _not_ have mapped the whole local map
+			{
+				if(player.HasMapped(mapSize))
+					return false;
+			}
+			else
+			// Condition is to have mapped at least the given amount of systems
+			{
+				if(!player.HasMapped(it.second))
```

Comment:
This should use the map size of the given outfit, not an arbitrary number.

Reply:
Done!